client/my-sites/plans/hooks/use-plan-prices.ts

```diff
@@ -0,0 +1,36 @@
+import { getPlan, PlanSlug } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { getPlanRawPrice, getDiscountedRawPrice } from 'calypso/state/plans/selectors';
+import { getPlanDiscountedRawPrice } from 'calypso/state/sites/plans/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export interface PlanPrices {
+	rawPrice: number;
+	discountedRawPrice: number; // discounted on yearly-monthly conversion
+	planDiscountedRawPrice: number; // discounted on site plan upgrade
+}
+
+interface Props {
+	planSlug: PlanSlug;
+	monthly?: boolean; // defaults to true
```

Comment:
It's not clear what this param does. Can you please explain or add a comment?

Reply:
Good point. It's the boolean passed to the selectors below, akin to `isMonthly`. They in turn fetch full-term or monthly-projected prices from what I can tell.

I prefer "monthly", I believe mainly due to not having a reference to something/"it" that is monthly or not. It's just a computation projection. The real semantic I guess would have been "projectToMonthly".

I see `getPlanDiscountedRawPrice` is using `isMonthly`, which I would have also considered for refactoring in similar vein 🤔

Reply:
Let me know if you prefer renaming to something and will happily make the change
I'm actually inclined to go with `projectToMonthly` now 🤔

Reply:
> I see getPlanDiscountedRawPrice is using isMonthly, which I would have also considered for refactoring in similar vein

I saw the same thing but before I read this comment. In https://github.com/Automattic/wp-calypso/pull/75140 I'm suggesting changing it to `returnMonthly`; what do you think? I also left a comment to explain what it does (and the risks of relying on it).

Reply:
Thanks, @sirbrillig. `returnMonthly` works for me. I can use the same here.

Reply:
`returnMonthly` sounds good to me!